### DISCUSSION TITLE
Real colored W11 start menu button in w10 taskbar style?

### DISCUSSION THREAD

**Discussion by zoosterix** [valinet/ExplorerPatcher#808] [Ideas]
There is W11 start button icon in W10 taskbar style, and it's fine, but it's not the actual W11 start button. It's small and white. Real W11 start button is a little bit bigger and it's blueish with a slight light gradient. Is it possible to have real W11 start button option in w10 taskbar style?

**Comment by valinet** [comment]
Ofc it's not the actual button, Microsoft never bothered to update the old taskbar. With the current implementation, that's it. At least regarding the size. That's the maximum extent `explorer` offers for drawing the button. Ofc, by maybe tweaking the code it would allow to draw in a bigger area, but that require tons of testing, making sure it doesn't broke something else, etc. It's not my code, and is not even code, stuff in `explorer` is mad eby Microsoft and I just have access as far as whatever information can be inferred from the disassembly. As for the colors, if someone spents the time to craft an algorithm that draws the correct colors pixel by pixel, I can integrate that. As far as the current implementation is concerned, the button is drawn mathematically, so I avoid having to ship bitmaps/textures, and have vector support out of the box. But this is hard to do like this, manually, for things more complicated than 2 lines across some square.

An alternative would be to use something like OpenShell which comes with its own button and use there whatever texture you'd like.

**Reply by Anixx** [reply]
By the way, in Classic theme the button is just a black rectangle if not to replace it with OpenShell:
![изображение](https://user-images.githubusercontent.com/1271736/152850397-27c47b6c-558c-4d24-a916-4477da52918b.png)

**Reply by valinet** [reply]
@Anixx Only Windows 11 button or Windows 10 button as well? And yeah, not really tested under classic theme.

**Reply by Anixx** [reply]
@valinet Windows 10 button as well. No button at all in any case.
